src/main/java/io/supertokens/storage/mysql/Start.java

```diff
@@ -934,4 +911,315 @@ public void setJWTSigningKey_Transaction(TransactionConnection con, JWTSigningKe
+        }
+    }
+
+    @Override
+    public PasswordlessCode[] getCodesOfDevice(String deviceIdHash) throws StorageQueryException {
+        try {
+            return PasswordlessQueries.getCodesOfDevice(this, deviceIdHash);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public PasswordlessCode[] getCodesBefore(long time) throws StorageQueryException {
+        try {
+            return PasswordlessQueries.getCodesBefore(this, time);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public PasswordlessCode getCode(String codeId) throws StorageQueryException {
+        try {
+            return PasswordlessQueries.getCode(this, codeId);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public PasswordlessCode getCodeByLinkCodeHash(String linkCodeHash) throws StorageQueryException {
+        try {
+            return PasswordlessQueries.getCodeByLinkCodeHash(this, linkCodeHash);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public io.supertokens.pluginInterface.passwordless.UserInfo getUserById(String userId)
+            throws StorageQueryException {
+        try {
+            return PasswordlessQueries.getUserById(this, userId);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public io.supertokens.pluginInterface.passwordless.UserInfo getUserByEmail(String email)
+            throws StorageQueryException {
+        try {
+            return PasswordlessQueries.getUserByEmail(this, email);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public io.supertokens.pluginInterface.passwordless.UserInfo getUserByPhoneNumber(String phoneNumber)
+            throws StorageQueryException {
+        try {
+            return PasswordlessQueries.getUserByPhoneNumber(this, phoneNumber);
+        } catch (SQLException e) {
+            throw new StorageQueryException(e);
+        }
+    }
+
+    @Override
+    public void createDeviceWithCode(@Nullable String email, @Nullable String phoneNumber, @NotNull String linkCodeSalt,
+            PasswordlessCode code) throws StorageQueryException, DuplicateDeviceIdHashException,
+            DuplicateCodeIdException, DuplicateLinkCodeHashException {
+        if (email == null && phoneNumber == null) {
+            throw new IllegalArgumentException("Both email and phoneNumber can't be null");
+        }
+        try {
+            PasswordlessQueries.createDeviceWithCode(this, email, phoneNumber, linkCodeSalt, code);
+        } catch (StorageTransactionLogicException e) {
+            // TODO: review these error messages
```

Comment:
If this is done, then remove this TODO. Also, I thought you were going to use regex matching - why was that removed?

Reply:
We've discussed that it may be better for consistency if the checked worked the same way as it does right now and to discuss and maybe update the way we check exceptions separately.

Reply:
okay!